### PR TITLE
item: fix item availability status.

### PR DESCRIPTION
**Pull request by zannkukai**
Closes rero/rero-ils#2170.
Closes rero/rero-ils#2240.

Co-authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Dependencies

* rero/rero-ils#2634.

## How to test?
Item detail (pro inteface)
image

Interface PRO (issue)
image

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: projects/admin/src/app/record/detail-view/document-detail-view/holding/default-holding-item/default-holding-item.component.ts [hunk 1]
```diff
@@ -45,12 +45,20 @@ export class DefaultHoldingItemComponent implements OnInit {
...
 
+  // GETTER & SETTER ==========================================================
   /** Current interface language */
   get language() {
     return this._translateService.currentLang;
   }
 
-  // CONSTRUCTOR & HOOKS ==============================================================
+  /** Get the number of pending request on this item */
+  get itemRequestCounter() {
```
Comment (jma) [inline projects/admin/src/app/record/detail-view/document-detail-view/holding/default-holding-item/default-holding-item.component.ts:55]: Can you add `@returns` documentation?

File: projects/admin/src/app/record/detail-view/document-detail-view/holding/default-holding-item/default-holding-item.component.ts [hunk 2]
```diff
@@ -45,12 +45,20 @@ export class DefaultHoldingItemComponent implements OnInit {
...
+    return (this.item.metadata.availability && this.item.metadata.availability.request)
+      ? this.item.metadata.availability.request
```
Comment (jma) [inline projects/admin/src/app/record/detail-view/document-detail-view/holding/default-holding-item/default-holding-item.component.ts:57]: What is `request` an array an integer?

File: projects/admin/src/app/record/detail-view/holding-detail-view/serial-holding-detail-view/serial-holding-detail-view.component.html [hunk 3]
```diff
@@ -108,7 +108,7 @@ <h1 class="mb-2">
                 *ngFor="let item of this.receivedItems"
                 [ngClass]="{'deleted-status': item.metadata.issue.status == 'deleted'}">
               <div class="col-sm-4 text">
-                <admin-record-masked *ngIf="item.metadata._masked" [record]="holding"></admin-record-masked>
+                <admin-record-masked *ngIf="item.metadata._masked" [record]="item"></admin-record-masked>
```
Comment (jma) [inline projects/admin/src/app/record/detail-view/holding-detail-view/serial-holding-detail-view/serial-holding-detail-view.component.html:111]: Is it a fix? Maybe you can add an entry in the commit message.

File: projects/admin/src/app/service/items.service.ts [hunk 4]
```diff
@@ -113,6 +113,21 @@ export class ItemsService {
     );
   }
 
+  /**
+   * Get item availability flag
+   * @param itemPid - the item pid
+   * @returns An Observable representing the item availability
+   */
+  getAvailability(itemPid: string): Observable<boolean> {
+    const url = `/api/item/${itemPid}/availability`;
```
Comment (jma) [inline projects/admin/src/app/service/items.service.ts:122]: Do not use an hard coded url, please use `apiService`.

**Review by jma — CHANGES_REQUESTED** [review]

File: projects/admin/src/app/record/detail-view/holding-detail-view/serial-holding-detail-view/serial-holding-detail-view.component.html (hunk 3 above)
Comment (zannkukai) [reply projects/admin/src/app/record/detail-view/holding-detail-view/serial-holding-detail-view/serial-holding-detail-view.component.html:111]: yes, this is the main fix about the related issue #2240.

File: projects/admin/src/app/record/detail-view/document-detail-view/holding/default-holding-item/default-holding-item.component.ts (hunk 1 above)
Comment (zannkukai) [reply projects/admin/src/app/record/detail-view/document-detail-view/holding/default-holding-item/default-holding-item.component.ts:55]: Method isn't yet used. I removed it

File: projects/admin/src/app/record/detail-view/document-detail-view/holding/default-holding-item/default-holding-item.component.ts (hunk 2 above)
Comment (zannkukai) [reply projects/admin/src/app/record/detail-view/document-detail-view/holding/default-holding-item/default-holding-item.component.ts:57]: `metadata.availability` is an object regrouping all necessary information to display (as well as possible) the item availability :
```json
{
  "available": false,
  "display_text": [
    {
      "label": "on_shelf",
      "language": "default"
    }
  ],
  "request": 1,
  "status": "on_shelf"
}
```

These informations are dumped from backend serialize.
So the `request` field is the number of pending request related to the item. It is always present into object.

**Review by jma — APPROVED** [review]

**Pull request merged**